### DISCUSSION TITLE
A Real Experience of Astroid Framework vs Helix Ultimate 2

### DISCUSSION THREAD

**Discussion by sparkktv** [templaza/astroid-framework#382] [General]
I used Astroid back when JoomDev ran it. And I loved Astroid, was very sad when JoomDev shut down. JoomDev helped get my website running witH Astroid due to a bug with YooTheme WidgetKit & Zoo.

When JoomDev went under I moved over to Helix Framework, which worked out of the box and most of the customizations I used with Astroid were very easy to port over to Helix. It was very interchangeable.

I recently found out that Astroid was resurrected by TemPlaza. So I started playing with it, and found out the fix JooMDev made for me no longer worked with Joomla 4. So @sonvnn was very nice enough to create a new fix and make Zoo & Widgetkit work with Astroid once again. Not really sure why they never worked in the first place with Astroid but after testing the fix for a few hours I started to maybe figure out why.

I was testing Astroid in my localhost setting using the fix supplied in the developer build of Astroid. With Zoo & Widgetkit, it makes Astroid painfully slow. For some reason Astroid tries to load the cache images from Widgetkit & Zoo instead of the direct images causing some images to no longer load because the cache images are cleared throughout the day (this was an old load method of Widgetkit but now most load direct). The average load time of my site was around 17 seconds using Astroid, compared to under 1 second on the default Helix Ultimate template.
The other issue is the fix supplied only works with Template Zero. For some reason Template One is still broken. I enabled JCH Optimize and the load was even slower on Astroid, taking over 24 seconds to load. And that even broke the images from Widgetkit totally. 
(And it's not the typography page as previously mentioned in an issue because I turned it off and loaded my own fonts from custom CSS and I also got rid of every template position except the ones I needed to load)

Helix Ultimate 2 loads quick out of the box, and is very customizable. And works great with YooTheme extensions and works the way those extensions were intended to work. The images load directly, enables lazyload and loads instantly.

I don't know what is slowing down Astroid or why the YooTheme extensions are such an issue with Astroid. I will say Astroid is a more advanced Framework with way more options, but most of those options are available in Helix via custom CSS. For now I have to stick with Helix, which is sad because I really do love Astroid and was happy to see it return. But with Joomla 4 it is way to slow for my website. Maybe Astroid needs re-written for J4 optimization since it was originally a J3 framework? I believe the cause is Widgetkit & Zoo but why are those slowing down Astroid but not other Frameworks? There is something or atlas has to be something in the framework not written properly that is causing these massive slowdowns. 
But I will keep my eye on Astroid and test it ever so often to see if real improvements have been made. I want to return to Astroid but just not possible at this time due to my experience. Again I appreciate the team at TemPlaza for their help and dedication to this project. I hope it can improve over time.  My eye is definitely on Astroid and this project.

**Comment by sonvnn** [comment]
Thanks for interest! Let's wait for Astroid 3. That will be the big answer for Helix Ultimate :)

**Comment by paurray** [comment]
@sonvnn Any rough idea of when Astroid 3 is out? Sounds exciting.

**Reply by sonvnn** [reply]
It's will ready for testing in the end of September.

**Reply by paurray** [reply]
Please let me know in advance and give me early access. Maybe we can do some "What is New in Astroid 3 Videos" paul(at)finalbug.tv
